File: samples/scoping-samples/simple-stack-example-scoping-java/src/main/java/com/zhuinden/simplestackexamplescoping/core/navigation/BaseKey.java
package com.zhuinden.simplestackexamplescoping.core.navigation;

import com.zhuinden.simplestack.ServiceBinder;
import com.zhuinden.simplestackextensions.fragments.DefaultFragmentKey;
import com.zhuinden.simplestackextensions.services.DefaultServiceProvider;

import javax.annotation.Nonnull;

public abstract class BaseKey
        extends DefaultFragmentKey
        implements DefaultServiceProvider.HasServices {
    @Nonnull
    @Override
    public String getFragmentTag() {
        return toString();
    }

    @Nonnull
    @Override
    public String getScopeTag() {
        return toString();
    }

    @Override
    public void bindServices(@Nonnull ServiceBinder serviceBinder) {
    }
}

File: samples/scoping-samples/simple-stack-example-scoping-kotlin/src/main/java/com/zhuinden/simplestackexamplescoping/core/navigation/BaseKey.kt
package com.zhuinden.simplestackexamplescoping.core.navigation

import com.zhuinden.simplestackextensions.fragments.DefaultFragmentKey
import com.zhuinden.simplestackextensions.services.DefaultServiceProvider

/**
 * Created by Zhuinden on 2018.09.17.
 */
abstract class BaseKey : DefaultFragmentKey(), DefaultServiceProvider.HasServices {
    override fun getFragmentTag(): String = toString()

    override fun getScopeTag(): String = fragmentTag
}
